src/caret_analyze/architecture/architecture.py

```diff
@@ -101,16 +101,65 @@ def is_target_comm(comm: CommunicationStructValue):
         return Util.find_one(is_target_comm, self.communications)
 
     def get_path(self, path_name: str) -> PathStructValue:
-        return self._path_manager.get_named_path(path_name)
+        if path_name not in self.path_names:
+            raise InvalidArgumentError(f'Failed to get named path. {path_name} not exist.')
+
+        named_path: PathStruct = Util.find_one(lambda x: x._path_name == path_name, self._paths)
+        return named_path.to_value()
 
     def add_path(self, path_name: str, path_info: PathStructValue) -> None:
-        self._path_manager.add_named_path(path_name, path_info)
+        if path_name in self.path_names:
+            raise InvalidArgumentError('Failed to add named path. Duplicate path name.')
+
+        child: List[Union[NodePathStruct, CommunicationStruct]] = []
+
+        for c in path_info.child:
+            if isinstance(c, NodePathStructValue):
+                from .graph_search import NodePathSearcher
+                node_path_searcher: NodePathSearcher = \
+                    NodePathSearcher(tuple(self._nodes), tuple(self._communications))
+
+                node_path: NodePathStruct = \
+                    node_path_searcher._find_node_path('' if c.subscribe_topic_name is None
+                                                       else c.subscribe_topic_name,
+                                                       '' if c.publish_topic_name is None
+                                                       else c.publish_topic_name, c.node_name)
```

Comment:
```suggestion
                def is_target_node(node: NodeStruct):
                    raise NotImplementedError('')

                def is_target_node_path(node_path: NodePathStruct):
                    raise NotImplementedError('')

                node: NodeStruct = Util.find_one(is_target_node, self._nodes)
                node_path: NodePathStruct = Util.find_one(is_target_node_path, node.paths)
                child.append(node_path)
```

Reply:
Fixed it in [6bcb719](https://github.com/tier4/CARET_analyze/pull/156/commits/6bcb71915e5197703e9e92ba06b09919e900d3f0)